src/widget/systemtrayicon.cpp

```diff
@@ -366,11 +381,15 @@ void SystemTrayIcon::setIcon(QIcon &icon)
         };
         QImage* image = new QImage(icon.pixmap(64, 64).toImage());
         if (image->format() != QImage::Format_RGBA8888_Premultiplied)
-        *image = image->convertToFormat(QImage::Format_RGBA8888_Premultiplied);
+            *image = image->convertToFormat(QImage::Format_RGBA8888_Premultiplied);
         GdkPixbuf* pixbuf = gdk_pixbuf_new_from_data(image->bits(), GDK_COLORSPACE_RGB, image->hasAlphaChannel(),
                                  8, image->width(), image->height(),
                                  image->bytesPerLine(), callbackFreeImage, image);
+        if(!pixbuf) // If pixbuf initialization fails, it won't trigger the callback.
```

Comment:
It should, but valgrind analysis suggests otherwise. I suppose the *correct* approach would be to switch to gtk-specific ```GBytes``` in place of ```QImage```, as you noted above, then we could use [```gdk_pixbuf_new_from_bytes```](https://developer.gnome.org/gdk-pixbuf/stable/gdk-pixbuf-Image-Data-in-Memory.html#gdk-pixbuf-new-from-bytes) and forgo the hassle of freeing in a callback and that all. I'll look into that as well.